src/phpDocumentor/ConfigurationFactory.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace phpDocumentor;
+
+final class ConfigurationFactory
+{
+    /**
+     * @var Uri
+     */
+    private $uri;
+
+    /**
+     * @var \SimpleXMLElement
+     */
+    private $xml;
+
+    /**
+     * @var string
+     */
+    private $schemaPath;
+
+    /**
+     * @var bool
+     */
+    private $validateUri;
+
+    /**
+     * @param Uri    $uri
+     * @param string $schemaPath
+     */
+    public function __construct(Uri $uri, $schemaPath = '')
+    {
+        if ($schemaPath === '') {
+            $schemaPath = __DIR__ . '/../../data/xsd/phpdoc.xsd';
+        }
+
+        $this->replaceLocation($uri);
+        $this->schemaPath = $schemaPath;
+    }
+
+    /**
+     * Replaces the location of the configuration file if it is different.
+     *
+     * @param Uri $uri
+     */
+    public function replaceLocation(Uri $uri)
+    {
+        if ($this->uri !== $uri) {
+            $this->validateUri = true;
+        }
+
+        $this->uri = $uri;
+    }
+
+    /**
+     * Converts the phpDocumentor configuration xml to an array.
+     *
+     * @return array
+     */
+    public function get()
+    {
+        $this->validate($this->uri);
+
+        $version = $this->checkIfVersionAttributeIsPresent($this->xml);
+        if ($version) {
+            $this->validateXmlStructure($this->xml);
+            $xml = new Phpdoc3XmlToArrayConverter($this->xml);
```

Comment:
I'm having trouble finding the correct strategy for doing this. One problem is that the phpdocumentor3 has a constructor argument, and the phpdocumentor2 strategy doesn't. 
I suppose having the version attribute is the case for ```match()```, but I can't seem to get the whole working.